### ISSUE TITLE
Interstate75W & OneWire - screen flickers on read

### ISSUE THREAD

**Issue by AdiGitalEU** [pimoroni/pimoroni-pico#828]
When using onewire and ds18x20 to read temperature, the whole screen flickers like the transmission blocks refreshing. Not sure if I'm doing something wrong but I thought the state machines run independently.

**Comment by AdiGitalEU** [comment]
It's machine.Pin(26)

**Comment by AdiGitalEU** [comment]
I'm using I2C on pins 26,27 (GP20, GP21). OneWire is using pin 31 (GP26). Both work fine (except sporadic CRC errors on DS18x20). It's only when the the read on 1Wire occurs, the screen flickers (stops displaying for fraction of a second). No issues when reading from I2C.

**Comment by AdiGitalEU** [comment]
Yeah, I've checked schematics for free pins as I need to add I2C, 1Wire, and serial. I checked them all if there is any conflict with I75, and there doesn't seem to be. So I started drawing some content on the panel. All looks fine except that flicker.

**Comment by alphanumeric007** [comment]
If you disable the one wire, code wise, does the flicker stop?

**Comment by AdiGitalEU** [comment]
It does not flicker when there is no read from 1Wire. I does it only when the actual transaction happens and it's only for that split of a second - however long the transmission lasts. It's like a blink, once per read.

**Comment by alphanumeric007** [comment]
Posting your code, with the <> add code option, may help.

**Comment by AdiGitalEU** [comment]
This is the simplest form to replicate the issue:

`from interstate75 import Interstate75, DISPLAY_INTERSTATE75_64X32
import time, onewire, ds18x20

i75 = Interstate75(display=DISPLAY_INTERSTATE75_64X32, color_order = 1)
gfx = i75.display

gray = gfx.create_pen(128,128,128)

gfx.set_pen(gray)
gfx.rectangle(0,0,63,31)

i75.update()

ds_pin = machine.Pin(22)
ds_sensor = ds18x20.DS18X20(onewire.OneWire(ds_pin))
roms = ds_sensor.scan()
print('Found DS18B20 : ', roms)

while True:
  ds_sensor.convert_temp()
  time.sleep_ms(750)
  for rom in roms:
    print("Temp ", ds_sensor.read_temp(rom))`

**Comment by alphanumeric007** [comment]
Pin 22 (physical pin 22) is LED_G, according to the schematic.

**Comment by AdiGitalEU** [comment]
`machine.Pin(22) `
refers to GP, not physical pin. If you connect the sensor to physical pin 29 the code works.

**Comment by alphanumeric007** [comment]
Are you sure about that? I thought it referred to the physical pin number?
I must have misunderstood it?
https://docs.micropython.org/en/latest/library/machine.Pin.html

**Comment by AdiGitalEU** [comment]
> Are you sure about that? I thought it referred to the physical pin number?
> I must have misunderstood it?
> https://docs.micropython.org/en/latest/library/machine.Pin.html

The hint is in the first example: Pin 0. There is no physical pin 0.

**Comment by alphanumeric007** [comment]
Ah, OK, I goofed and got it the wrong way around. I'm back to I don't know what your issue is. =(
